python/ray/dataframe/dataframe.py

```diff
@@ -2882,35 +2881,72 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
                     are the quantiles.
         """
 
-        def quantile_helper(df, q, axis, numeric_only, interpolation):
-            try:
-                return df.quantile(q=q, axis=axis, numeric_only=numeric_only,
-                                   interpolation=interpolation)
-            except ValueError:
-                return pd.Series()
+        # deal with error checking for bad dtyping when numeric_only is False
+        if not numeric_only:
+            # check if there are any object columns
+            for t in self.dtypes:
+                if t == np.dtype('O'):
+                    e_msg = "Cannot call quantile on columns of type " + str(t)
+                    raise TypeError(e_msg)
+        else:
+            if all([dtype == np.dtype('O') or
+                    np.issubdtype(dtype, np.datetime64)
```

Comment:
Seems that `np.datetime64` type columns are supported.

```
In [8]: x
Out[8]: 
                 col1
0 2018-05-04 20:50:29
1 2018-05-04 20:50:29
2 2018-05-04 20:50:29

In [9]: x.dtypes
Out[9]: 
col1    datetime64[ns]
dtype: object

In [10]: x.quantile(q=0.5, axis=1, numeric_only=True)
Out[10]: 
0   NaN
1   NaN
2   NaN
Name: 0.5, dtype: float64
```

Reply:
Made a fix to handle all datetime objects